src/components/FollowButton/index.tsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react';
+import { useRecoilValue } from 'recoil';
+
+import { useCreateFollow, useDeleteFollow } from '@/hooks/useFolllow';
+import { useModal } from '@/hooks/useModal';
+import { userAtom } from '@/recoil/user';
+import { ModalType } from '@/types/modal';
+
+import { Button } from './style';
+
+export default function FollowButton(props: any) {
+  const [isMe, setIsMe] = useState(false);
+  const [isFollowing, setIsFollowing] = useState(false);
+  const user = useRecoilValue(userAtom);
+  const { openModal } = useModal();
+
+  const follower = props.followers.filter(
+    (follower: any) => follower.user === props.userId,
+  );
+
+  const { mutate: createFollowMutate } = useCreateFollow({
+    userId: props.userId,
+  });
+
+  const { mutate: deleteFollowMutate } = useDeleteFollow();
+
+  useEffect(() => {
+    if (props.userId == user?.email) setIsMe(true);
+    if (follower.length > 0) setIsFollowing(true);
+  }, []);
+
+  const viewFriends = async () => {
+    openModal({ type: ModalType.VIEW_FOLLOW });
+  };
+
+  const follow = () => {
+    createFollowMutate(props.userId);
+    setIsFollowing(true);
+  };
```

Comment:
위의 내용과 이어지는 내용입니닷~

unfollow를 하려면 follow 데이터의 id가 필요하기 때문에
**follow후에 API로부터 받은 data를 followData로 설정**해주어야 해요!
그래야 버튼을 토글할 수 있더라구요

createFollowMutate의 response 값을 비동기로 받아와야 하기 때문에
`mutate를 mutateAsync로 변경`하고 `async-await`을 사용해주시면 잘 동작할 것 같습니다 ㅎㅎ
```suggestion

  const { mutateAsync: createFollowMutate } = useCreateFollow({
    userId,
  });
  
  const follow = async () => {
    const followData = await createFollowMutate(userId);
    setFollowData(followData);
  };
```